tests/model_fields/models.py

```diff
@@ -578,3 +578,13 @@ class GeneratedModelNullVirtual(models.Model):
 
     class Meta:
         required_db_features = {"supports_virtual_generated_columns"}
+
+
+class Square(models.Model):
+    side = models.IntegerField()
+    area = models.GeneratedField(
+        expression=F("side") * F("side"),
+        output_field=models.BigIntegerField(),
+        db_persist=True,
+    )
+    fk = models.ForeignKey(Foo, on_delete=models.CASCADE)
```

Comment:
This will crash on databases without support for `GeneratedField`:
```
class Meta:
    required_db_features = {"supports_stored_generated_columns"}
```
Can we instead add nullable foreign keys to existing models? (e.g.  `GeneratedModel` and `GeneratedModelVirtual`.)